port/unix/omrsysinfo.c

```diff
@@ -1585,8 +1594,15 @@ omrsysinfo_shutdown(struct OMRPortLibrary *portLibrary)
 			PPG_si_executableName = NULL;
 		}
 #if defined(LINUX)
+		omrthread_monitor_enter(cgroupEntryListMonitor);
 		freeCgroupEntries(portLibrary, PPG_cgroupEntryList);
+		omrthread_monitor_exit(cgroupEntryListMonitor);
```

Comment:
The monitor needs to protect the nulling of the ` 		PPG_cgroupEntryList = NULL;` as well as the call to freeCgroupEntries()